### PR TITLE
Add code documentation

### PR DESCRIPTION
**Pull request by JakubBilski**
I followed this www.datacamp.com/community/tutorials/docstrings-python tutorial.
What I wrote can be used to automatically create a documentation in html format.
Example (our main.py):
docsSample
I updated readme with instructions on how to use this generator

### CODE REVIEW

File: src/music/tonation.py [hunk 1]
```diff
@@ -3,6 +3,18 @@
 
 
 class Tonation(base_sound.BaseSound):
+    """A sound of one pitch present in a music piece
+
+    Attributes:
+        kind (str) : "minor" or "major"
+        tonic (Chord) : Set of sounds composing the
```
Comment (JakubBrojacz) [inline src/music/tonation.py:10]: I feel like we can delete toniv, subdominant and dominant params. It is only used in bar_algorithm and we can remove it (or even leave just daria)

**Review by JakubBrojacz — CHANGES_REQUESTED** [review]

File: src/music/tonation.py (hunk 1 above)
Comment (JakubBilski) [reply src/music/tonation.py]: Removed

File: src/sounds_manipulation.py [hunk 2]
```diff
@@ -2,6 +2,17 @@
 
 
 def change_tonation(sounds, diff):
+    """Transpose all sounds a chosen number
+    of halftones up
+
+    Parameters:
+    sounds (list[Sound]) : Sounds to transpose
+    diff (int) : Chosen number of halftones
+    duration_ms_of_32 (float) : Duration of 32th note in miliseconds
```
[the remark this answers is not in the document]
Comment (JakubBilski) [reply src/sounds_manipulation.py]: Fixed

File: src/main.py [hunk 3]
```diff
@@ -48,6 +48,7 @@ def parse_args():
 
 @app.route('/recorded', methods=['POST'])
 def frontend_communication_upload_recorded():
+    """Used to upload a previously recorded audio file"""
```
[the remark this answers is not in the document]
Comment (JakubBilski) [reply src/main.py]: Added, I didn't go too much into details, hope it's enough

File: src/chords_simplification.py [hunk 4]
```diff
@@ -38,6 +38,22 @@
 
 
 def simplify(notes, chords, tonation, preview_file):
+    """Transpose all piece information to a key
+    in which it should be easier to play
```
[the remark this answers is not in the document]
Comment (JakubBilski) [reply src/chords_simplification.py]: the piece
I rewrote it, I think it's better now

File: src/chords_simplification.py [hunk 5]
```diff
@@ -38,6 +38,22 @@
...
+    notes (list[Sound])
+    chords (list[Chord])
+    tonation (Tonation)
+    preview_file (os.path-like)
+
+    Returns:
+    (list[Sound]) : Notes after transposition
+    (list[Chord]) : Chords after transposition
+    (Tonation) : Chosen easier key
+    (os.path-like) : Same as input preview_file
```
[the remark this answers is not in the document]
Comment (JakubBilski) [reply src/chords_simplification.py]: Fixed

File: src/chords_generation/daria_params.py [hunk 6]
```diff
@@ -1,3 +1,4 @@
+"""Global arameters used by chords generation functions"""
```
[the remark this answers is not in the document]
Comment (JakubBilski) [reply src/chords_generation/daria_params.py]: Fixed

**Comment by JakubBilski** [comment]
Ok, so it turns out, that if something is a part of a package, you should never run it as a script.
https://stackoverflow.com/questions/16981921/relative-imports-in-python-3
That's the whole reason it was so hard to get tests to work, and thet's the reason we needed path_magic. I moved main and main_tests outside the src folder, and changed a lot of imports (if not all of them) to relative. Now the whole app still works, we don't need path_magic, and authomatic documentation also works (it's in the html folder).
I would love to change music_server.py name to sth else, but I didn't have a good idea.

**Pull request merged**